R/z_pages.R

```diff
@@ -24,6 +24,21 @@
 #'
 #' @export
 animint2pages <- function(plot.list, github_repo, commit_message = "Commit from animint2pages", private = FALSE, ...) {
+  # Check for GitHub token
+  github_token <- Sys.getenv("GITHUB_PAT")
+  if (identical(github_token, "")) {
```

Comment:
hi @Faye-yufan thanks for adding the error message, but are you sure this is the right condition to test? https://gitcreds.r-lib.org/reference/gitcreds_get.html does not say that it sets GITHUB_PAT so I'm not sure that is correct. I was thinking we could use gitcreds_get, but that man page says "gitcreds_get() errors if git is not installed, no credential helpers are configured or no credentials are found." so I think we would have to tryCatch that error. 
Looking below I see
```r
  if (is.null(whoami)) {
    stop("A GitHub token is required to create and push to a new repo.")
  }
```
Is this error different? Can they be combined?

Reply:
You're correct; the condition seems to be wrong.
I'm not sure these two can be combined or not, `gitcreds_get()` is more about fetching credentials stored in the systems's git. `gh::gh_whoami()` interacts with the GitHub API to get the information for the current GitHub user.
Still testing the functions.

Reply:
Update some analysis:
I just tested on a Windows laptop without Git installed, and I found using either `gitcreds_get()` or `gh_whoami()` for the validate condition is fine because if one works (returns with value and no error), the other would yield the same result. 

However, I suggest using `Sys.setenv()` to set the GITHUB_PAT and `gert::git_config_global_set()` to configure the user.name and user.email, like below:
```
Sys.setenv(GITHUB_PAT="yourGithubPAT")
gert::git_config_global_set("user.name", "yourUserName")
gert::git_config_global_set("user.email", "yourUser")
```
This approach is not requiring the Git system to be installed on the machine (`gitcreds_set()` requires), and simplifying the setup process for users. After setting these configs, users can push the repo without any other steps.